manifests/init.pp

```diff
@@ -17,7 +17,11 @@
 class cobald(
   String                     $cobald_version             = undef,                        # cobald version to be used (undef = latest PyPI release, 'master' = Github master branch or PyPI release number)
   String                     $tardis_version             = undef,                        # tardis version to be used (undef = latest PyPI release, 'master' = Github master branch or PyPI release number)
+  Enum['PyPI', 'git']        $cobald_provider            = 'PyPI',                     # Cobald source provider (PyPI or Git)
+  Enum['PyPI', 'git']        $tardis_provider            = 'PyPI',                     # TARDIS source provider (PyPI or Git)
   Array[Enum['krb5', 'ssh']] $auth_lbs                   = ['krb5'],                     # authentication used to access local batch system
+  String                     $cobald_git_url             = $cobald::params::cobald_url,  # Cobald Git URL
+  String                     $tardis_git_url             = $cobald::params::tardis_url,  # TARDIS Git URL
```

Comment:
What about calling the new variables `cobald_url` and `tardis_url` which default to `undef`? They could be used to specify either the git repository URL or the base URL of Python Package Index. Their meanings depend on the `cobald_repo_type` and the `tardis_repo_type` switches. If they are left to `undef`, the code is fetched from `git+https://github.com/MatterMiners/cobald` and `git+https://github.com/MatterMiners/tardis` in the `git` case and `https://pypi.org/simple` in the `pypi` case.

@stefan-k, @olifre: What do you think?

Reply:
I think that's a good idea, since also PyPI may use a different URL. Of course, that means that the code installing from PyPI needs to be extended to use that URL, too (and default to `https://pypi.org/simple`). It seems that works via the `index` class parameter of the `puppet-python` classes.

Reply:
Thanks for your suggestions. I named the new variables `cobald_repo_url` and `tardis_repo_url` because `cobald_url` and `tardis_url` are used in `install.pp` and cannot be overwritten (weird puppet...). I chose renaming the exposed parameters instead of the internal variables because I think this is more consistent with `cobald_repo_type` and `tardis_repo_type`.